Read `version` for `update_appstore_strings` from codebase

mokagio opened the pull request:
### Description
The `update_appstore_strings` required a `version` parameter at call site, but the information we passed to it – the current app version – could be read from the codebase itself.

Not having to pass the parameter makes the process simpler, _and more shell history friendly_.

### Testing instructions

Here's a screenshot of the lane running with no parameter and correctly picking up [11.2](https://github.com/woocommerce/woocommerce-ios/blob/c0439faccead6ef296176c6410d94f161558faec/config/Version.Public.xcconfig) as the version

![image](https://user-images.githubusercontent.com/1218433/202834031-55448823-02b3-4a80-9c3f-98f96169eb89.png)

Here's a screenshot of the lane running with `version: 1.2.3` to override the one in the code. You can see the lane uses it and updates the `.po` file. The push then failed because I hadn't `git push -u` my branch yet.

![image](https://user-images.githubusercontent.com/1218433/202834058-073d6380-3d13-4531-8468-869c3b087568.png)


---

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

mokagio committed:
Read `version` for `update_appstore_strings` from codebase

oguzkocer reviewed (approved):
Thank you for this improvement @mokagio!

---

I tested this by:

1. Created a new branch from `mokagio/remove-version-update-asc-strings`
2. Merged `release/11.3` into it
3. Called `bundle exec fastlane update_appstore_strings`

As expected, a new commit with the following diff was created:

```
﻿diff --git a/WooCommerce/Resources/AppStoreStrings.pot b/WooCommerce/Resources/AppStoreStrings.pot
index ab943c15da..d69c7b95a1 100644
--- a/WooCommerce/Resources/AppStoreStrings.pot
+++ b/WooCommerce/Resources/AppStoreStrings.pot
@@ -57,9 +57,12 @@ msgctxt "app_store_promo_text"
 msgid "Run your store from anywhere"
 msgstr ""
 
-msgctxt "v11.2-whats-new"
+msgctxt "v11.3-whats-new"
 msgid ""
-"For this release, we focused on polishing a few things and taking care of minor bugs that our team found, including the option to preview your products before publishing them. Don’t be fooled – we are working on some pretty cool stuff for you all! Keep sharing feedback 💜\n"
+"- [*] In-Person Payments: Show spinner while preparing reader for payment, instead of saying it's ready before it is. [https://github.com/woocommerce/woocommerce-ios/pull/8115]\n"
+"- [internal] In-Person Payments: update StripeTerminal from 2.7 to 2.14 [https://github.com/woocommerce/woocommerce-ios/pull/8132]\n"
+"- [*] In-Person Payments: Fixed payment method prompt for WisePad 3 to show only Tap and Insert options [https://github.com/woocommerce/woocommerce-ios/pull/8136]\n"
+"\n"
 msgstr ""
 
 #. translators: This is a promo message that will be attached on top of a screenshot in the App Store.
```

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -217,10 +217,10 @@ platform :ios do
   # -----------------------------------------------------------------------------------
   # Usage:
-  # bundle exec fastlane update_appstore_strings version:<release note version>
+  # bundle exec fastlane update_appstore_strings
```
Technically the option is still available, just optional now, so I'd document it like this maybe?

```suggestion
  # bundle exec fastlane update_appstore_strings [version:…]
```

Alternatively we could use the occasion to convert the documentation of this lane to YARD docs like we now do in other repos (converting the doc of all the lanes in that repo is probably out of scope of this PR, but converting just this one could be a good start even if we don't yet update the others)

AliSoftware reviewed (approved):
Approved with documentation nitpick, otherwise lgtm 👍

mokagio committed:
Convert `update_appstore_strings` documentation to YARD-style

See conversation at
https://github.com/woocommerce/woocommerce-ios/pull/8160#discussion_r1027081798

This is the documentation format we should have going forward, and can
be seen rolled out more widely in WordPress iOS, e.g.:
https://github.com/wordpress-mobile/WordPress-iOS/blob/784c2a4f2ad74deb36da0a20d84b6fdd16ed56b6/fastlane/lanes/localization.rb#L149-L163

Updating the entire `Fastfile` is out of scope for the piece of work
this change belongs to, #8160. Hopefully this can be the first in an
organic migration process, which we could then complete once critical
mass has been reached with a focused burst.
